### DISCUSSION TITLE
Feature: Height-restricted scenery scatter

### DISCUSSION THREAD

**Discussion by ocalhoun6** [OpenRCT2/OpenRCT2#15733] [Feature requests]
Kind of related to #10825 -- but I figured it should be its own separate issue.

I'm thinking that 99% of the time when you're using the scenery scatter tool near a cliff, you _don't_ want it to place scenery on land that's far above or far below the level where you're actively working.

Don't worry -- I'm not asking for extra buttons this time! I just think that the scenery tool should consider the median level of the land/water surface it's placing items on, then refuse to place any items on surfaces (say) 15 height units higher or lower than that ... so when you're using the scatter tool near a cliff, it's easy to only place items at the top or bottom of the cliff, even if the cliff face is irregularly shaped.

Now that the scenery scatter tool is resizable, I think this would make even more sense. One place where it might break down, for example, is places where you'd want to scatter things on the top of a very narrow tower or ledge, but with a large scatter tool, it would always only place things on the lower surface around your narrow cliff. (or vice versa for a narrow hole or trench) But since the tool is resizable now, in a case like that you could just make the scatter tool smaller and use it only at the top of your narrow cliff.

And, of course, if you _do_ want it to scatter things at both the top and the bottom of the cliff, you can reduce the size of the tool, scatter a few times on the top and a few times on the bottom.

But I do think that in the vast majority of cases when you're using the scatter tool near a cliff, you want it to only scatter at the top or bottom of the cliff, not both at once.

**Comment by Ruedii** [comment]
I'd also like to see several of these premade for common use cases.
